backend/src/controllers/MyRestaurantController.ts

```diff
@@ -0,0 +1,32 @@
+import { Request, Response } from 'express';
+import Restaurant from '../models/restaurant';
+import mongoose from 'mongoose';
+import { v2 as cloudinary } from 'cloudinary';
+
+const createMyRestaurant = async (req: Request, res: Response) => {
+  try {
+    const existingRestaurant = await Restaurant.findOne({ user: req.userId });
+    if (existingRestaurant) {
+      return res.status(409).json({ message: 'Restaurant already exists' });
+    }
+
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/omkardongre/Food-Ordering-App/pull/27#issuecomment-2383391092>#6fe9f8</a><span> - 09/30/2024, 02:40 pm</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> Lack of file type and size validation before processing the uploaded image
</div>
<div id="fix">
<b>Fix:</b> Implement file type and size validation before processing the uploaded image. Use a library like 'express-validator' or custom middleware to check the file's MIME type and size.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -10,6 +10,15 @@
      return res.status(409).json({ message: 'Restaurant already exists' });
    }

 +  const image = req.file as Express.Multer.File;
 +  if (!image) {
 +    return res.status(400).json({ message: 'No image file uploaded' });
 +  }
 +
 +  const allowedTypes = ['image/jpeg', 'image/png', 'image/gif'];
 +  const maxSize = 5 * 1024 * 1024; // 5MB
 +
 +  if (!allowedTypes.includes(image.mimetype) || image.size > maxSize) {
 +    return res.status(400).json({ message: 'Invalid file type or size' });
 +  }
 +
    const base64Image = image.buffer.toString('base64');
    const dataURI = 'data:${image.mimetype};base64,${base64Image}';
```
</div>
</details>
</div>
<br>


</div>

Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged

Reply:
size validation is done in multer configuration